crates/valence_new/src/inventory.rs

```diff
@@ -159,3 +162,108 @@ pub(crate) fn update_player_inventories(mut query: Query<(&mut Inventory, &mut C
         }
     }
 }
+
+/// Used to indicate that the client with this component is currently viewing
+/// an inventory.
+#[derive(Debug, Clone, Component)]
+pub struct OpenInventory {
+    /// The Entity that has the Inventory that the client is currently
+    /// viewing.
+    pub(crate) entity: Entity,
+}
+
+impl OpenInventory {
+    pub fn new(entity: Entity) -> Self {
+        OpenInventory { entity }
+    }
+}
+
+/// Handles the `OpenInventory` component being added to a client, which
+/// indicates that the client is now viewing an inventory.
+pub(crate) fn update_client_on_open_inventory(
+    mut clients: Query<(&mut Client, &OpenInventory, Added<OpenInventory>)>,
+    inventories: Query<&Inventory>,
+) {
+    for (mut client, open_inventory, _) in clients.iter_mut() {
+        // validate that the inventory exists
+        let inventory = inventories.get_component::<Inventory>(open_inventory.entity);
+        if inventory.is_err() {
+            warn!("Client is viewing an inventory that does not exist");
+            continue;
+        }
+
+        // send the inventory to the client
+        let inventory = inventory.unwrap();
```

Comment:
This can be better expressed with `let Ok(inventory) = ... else { ...; continue };` to avoid the `unwrap`.

Reply:
I actually need to rewrite most of this system because it doesn't account for multiple observers of the same inventory.